mapengine/worldwind/apk/src/main/java/mil/emp3/worldwind/utils/BoundingBoxGeneration.java

```diff
@@ -163,6 +166,38 @@ private boolean boundsForEntireGlobeVisible(IGeoBounds geoBounds, IGeoPosition g
         }
         return false;
     }
+
+    private boolean boundsForSmallCameraAngles(IGeoBounds geoBounds) {
+        if (null == camera) {
+            return false;
+        }
+
+        if (((Math.abs(camera.getHeading()) <= 5.0) || (Math.abs(camera.getHeading()) >= 355.0)) &&
+                (Math.abs(camera.getTilt()) <= 5.0) &&
+                (Math.abs(camera.getRoll()) <= 5.0)) {
+
+            IGeoPosition centerNorth = mapInstance.containerToGeo(new Point(width / 2, 0));
```

Comment:
It would be slightly more efficient to use (mapController.screenPointToGroundPosition) and avoid going via mapInstance. in MapInstance there is a check for UI thread and when in boundsGeneration we have already done it.

BoundsGeneration and BoundingBoxGeneration both use mapController directly, so for consistency this code can do the same.